tests/udp_validate_with_restpy/test_udp_header_with_counter_restpy.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+import utils
+from abstract_open_traffic_generator import flow as Flow
+
+
+@pytest.mark.parametrize('packets', [100])
+@pytest.mark.parametrize('size', [74])
+def test_udp_header_with_counter(api, b2b_raw_config, size, packets):
+    """
+    Configure a raw udp flow with,
+    - Non-default Counter Pattern values of src and
+      dst Port address, length, checksum
+    - 100 frames of 74B size each
+    - 10% line rate
+
+    Validate,
+    - tx/rx frame count is as expected
+    - all captured frames have expected src and dst Port address
+    """
+    flow = b2b_raw_config.flows[0]
+
+    src_port = ('5000', '2', '10')
+    dst_port = ('6000', '2', '10')
+    length = ('35', '1', '2')
+    checksum = ('6', '1', '2')
+    flow.packet = [
+        Flow.Header(
+            Flow.Ethernet(
+                src=Flow.Pattern('00:0c:29:1d:10:67'),
+                dst=Flow.Pattern('00:0c:29:1d:10:71')
+            )
+        ),
+        Flow.Header(
+            Flow.Ipv4(
+                src=Flow.Pattern('10.10.10.1'),
+                dst=Flow.Pattern('10.10.10.2')
+            )
+        ),
+        Flow.Header(
+            Flow.Udp(
+                src_port=Flow.Pattern(
+                    Flow.Counter(start=src_port[0], step=src_port[1],
+                                 count=int(src_port[2]))
+                ),
+                dst_port=Flow.Pattern(
+                    Flow.Counter(start=dst_port[0], step=dst_port[1],
+                                 count=int(dst_port[2]), up=False)
+                ),
+                length=Flow.Pattern(
+                    Flow.Counter(start=length[0], step=length[1],
+                                 count=int(length[2]))
+                ),
+                checksum=Flow.Pattern(
+                    Flow.Counter(start=checksum[0], step=checksum[1],
+                                 count=int(checksum[2]))
+                )
+            )
+        ),
+    ]
+    flow.duration = Flow.Duration(Flow.FixedPackets(packets=packets))
+    flow.size = Flow.Size(size)
+    flow.rate = Flow.Rate(value=10, unit='line')
+
```

Comment:
does these really required?

Reply:
Rate is not required but size and duration is.

Num of packets affects field pattern expansion. Size may sometimes be useful to check whether protocol headers result in size auto-adjust.

Reply:
@ashutshkumr Now the implementation is fetching the packet configuration from the packet fields instead of packet preview.
So, duration and size doesn't take any affect.

Reply:
@Rangababu-R you're right, since we're only validating against config and not the preview, they're not required.

But as an aside, as long as it doesn't affect the test run duration, I'm fine with keeping them. It helps with reusablitly (copy-paste without adding/removing). What do you think ?